renku/models/datasets.py

```diff
@@ -123,10 +123,12 @@ def from_commit(cls, commit):
         """Create an instance from a Git commit."""
         return cls(name=commit.author.name, email=commit.author.email)
 
-    def __attrs_post_init__(self):
-        """Post-Init hook to set _id field."""
-        if not self._id:
-            self._id = 'mailto:{self.email}'.format(self=self)
+    @_id.default
+    def default_id(self):
+        """Set the default id."""
+        if self.email:
+            return 'mailto:{self.email}'.format(self=self)
```

Comment:
I think we don't need to pass entire self context to format. I would just pass what's needed.

Reply:
true - fixed